torch/distributed/rpc/backend_registry.py

```diff
@@ -1,15 +1,20 @@
 from __future__ import absolute_import, division, print_function, unicode_literals
 
+import collections
+import enum
 
-_BACKEND_REGISTRY = {}
+import torch.distributed.distributed_c10d as dc10d
+from torch.distributed import ProcessGroupAgent
 
 
-def _get_backend_registry():
-    return _BACKEND_REGISTRY
+BackendValue = collections.namedtuple("BackendValue", ["init_backend_handler"])
 
+# Create an enum type, `BackendType`, with empty members.
+BackendType = enum.Enum(value="BackendType", names={})
 
-def is_backend_registered(backend_name):
-    return backend_name in _get_backend_registry()
+
+def _is_backend_name_registered(backend_name):
```

Comment:
Is this function just used once in `register_backend`? If that's the case, would it be better to make it inline?

Reply:
Yes, it's not an external API anymore. `backend_name` validity is tested when accessing a particular invariont of the enum type.

For this short one, it's ok to make it inline.